### PR TITLE
lint infinite loop

### LINKED ISSUE #769: lint infinite loop.
I have noticed when running the test server the lint goes into an infinite loop, perhaps it is related to this line below.  we watch for a script change, and then lint and the linting process changes the script etc.

https://github.com/yeoman/generator-webapp/blob/c62391b8a58237d9f52f2763497d2e6c049e5833/app/templates/gulpfile.js#L217

### PR DESCRIPTION
**Pull request by erniehs-zz**
Fixes #769.

### CODE REVIEW

**Comment by silvenon** [comment]
Thanks for the PR! While this technically solves the problem of the infinite loop, it also stops linting test files and leaves the task `lintTest` unused.

Instead I propose modifying `lintTest` to stop autofixing by making `lintBase` configurable:

```js
const lintBase = (files, options) => {
  return src(files)
    .pipe($.eslint(options)) // options are passed to ESLint
    // ...
}
function lint() {
  return lintBase('app/scripts/**/*.js', { fix: true }) // configure autofix
    .pipe(dest('app/scripts'));
};
// lintTest stays as-is
```

What do you think?

**Review by silvenon — CHANGES_REQUESTED** [review]
Thanks for the PR! While this technically solves the problem of the infinite loop, it also stops linting test files, which is the intention of `lintTest`, and leaves it unused.

Instead I propose modifying `lintTest` to stop autofixing by making `lintBase` configurable:

```js
const lintBase = (files, options) => {
  return src(files)
    .pipe($.eslint(options)) // options are passed to ESLint
    // ...
}
function lint() {
  return lintBase('app/scripts/**/*.js', { fix: true }) // configure autofix
    .pipe(dest('app/scripts'));
};
// lintTest stays as-is
```

What do you think?

**Comment by silvenon** [comment]
Actually, in `lintTest` we would have to lose the `dest` part to prevent writing the file:

```js
function lintTest() {
  return lintBase('test/spec/**/*.js');
};
```

**Comment by erniehs-zz** [comment]
Perhaps write the lint dest to tmp?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by silvenon** [comment]
It would be misleading to write files that we won't use. Is there a reason why you want to write them somewhere?

**Comment by erniehs-zz** [comment]
Not really I was thinking that they rewrote the files in the editor at times (during infinite loop)  however if loop is gone the fine 🙂

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by silvenon** [comment]
Great! Are you willing to add these changes to your PR? And leave the `watch` part.

**Comment by erniehs-zz** [comment]
Hola,

I will have a go implementing these changes

Cheers 🙂

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: app/templates/gulpfile.js [hunk 1]
```diff
@@ -213,8 +212,7 @@ function startTestServer() {
   });
 
   watch('app/scripts/**/*.js', scripts);
-  watch(['test/spec/**/*.js', 'test/index.html']).on('change', server.reload);
-  watch('test/spec/**/*.js', lintTest);
+  watch(['test/spec/**/*.js', 'test/index.html'], lintTest);
```
Comment (silvenon) [inline app/templates/gulpfile.js]: Why is this part changed? I suggest returning it to the way it was:

```suggestion
  watch(['test/spec/**/*.js', 'test/index.html']).on('change', server.reload);
  watch('test/spec/**/*.js', lintTest);
```

**Review by silvenon — CHANGES_REQUESTED** [review]

**Comment by erniehs-zz** [comment]
The browser refresh hppens twice as it occurs in lint base.  The server reload part is not needed here it seems.

Ernie

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

File: app/templates/gulpfile.js (hunk 1 above)
Comment (erniehs-zz) [reply app/templates/gulpfile.js]: see below.

Comment (silvenon) [reply app/templates/gulpfile.js]: You're correct that `lintBase` already reloads the server, so the `test/spec/**/*.js` pattern is indeed not needed here, but it doesn't reload when modifying `test/index.html` so I recommend keeping that pattern.

As for your second change, why would changing `test/index.html` need to trigger `lintTest`? Is that on purpose?

Comment (silvenon) [reply app/templates/gulpfile.js]: This is my updated code suggestion:

```suggestion
  watch('test/index.html').on('change', server.reload);
  watch('test/spec/**/*.js'], lintTest);
```

Comment (silvenon) [reply app/templates/gulpfile.js]: Alternatively, you can remove the `server.reload` part from `lintBase`, that doesn't make much sense anyway and it would probably make the reloading behavior more obvious.

I'm fine with either solution.

Comment (erniehs-zz) [reply app/templates/gulpfile.js]: splendid, let me suggest another change in line with this and see where it goes :) we're getting there :D

Comment (silvenon) [reply app/templates/gulpfile.js]: Thanks! ❤️ I hope this is not getting tedious for you.

Comment (erniehs-zz) [reply app/templates/gulpfile.js]: ha ha no probs ;)

File: app/templates/gulpfile.js [hunk 2]
```diff
@@ -212,8 +211,7 @@ function startTestServer() {
     }
   });
 
-  watch('app/scripts/**/*.js', scripts);
```
Comment (silvenon) [inline app/templates/gulpfile.js:215]: Please don't remove this.

**Review by silvenon — CHANGES_REQUESTED** [review]
Almost… 🤞😉

**Review by silvenon — APPROVED** [review]
Great!

**Comment by silvenon** [comment]
Thank you for spotting this crippling error. Published in [v4.0.0-8](https://github.com/yeoman/generator-webapp/releases/tag/v4.0.0-8). 🚀
